ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -87,6 +87,13 @@ internal class PaywallViewModelImpl(
         updateState()
     }
 
+    fun updateOptions(options: PaywallOptions) {
+        if (this.options != options) {
+            this.options = options
+            updateState()
```

Comment:
`updateState` only uses `options.offeringSelection`, so it seems silly to  fetch everything if we only change something else like `shouldDisplayDismissButton`.

Suggestion: compare only `options.offeringSelection` here, but the risk is that if `updateState` uses something else in the future, it would break.
So to deal with that, we can extract a `createState` method like `suspend createState(offeringSelection: OfferingSelection): PaywallState`.

Then make `updateState` simply do:
```kotlin
private fun updateState() {
    viewModelScope.launch {
         _state.value = createState(options.offeringSelection)
    }
}
```

And we can do the exact same here, therefore ensuring that if `createState` ever needs more data from `options`, that's explicit here.

What do you think?

Reply:
Actually, we also use the `mode` (it's masked behind a computed property in `PaywallViewModelImpl`), and `options.shouldDisplayDismissButton` in `calculateState`. So we would need those 3.

The other fields are the dismissRequest callback, the listener and the fontProvider... So I kinda think it's worth to just compare everything and if anything changes, just recompute the state... Wdyt?

Reply:
Oh that's right. This is fine then :+1: